cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -381,6 +378,32 @@ private CConfiguration createContainerCConf(CConfiguration cConf) {
     // container directory
     result.set(Constants.CFG_LOCAL_DATA_DIR, "data");
 
+    // Setup the configurations for isolated mode
+    if (clusterMode == ClusterMode.ISOLATED) {
+      // Disable implicit transaction
+      result.set(Constants.AppFabric.PROGRAM_TRANSACTION_CONTROL, TransactionControl.EXPLICIT.name());
+
+      // Disable explore
+      result.set(Constants.Explore.EXPLORE_ENABLED, Boolean.FALSE.toString());
+
+      // Bind the runtime monitor server to all interfaces
+      result.set(Constants.RuntimeMonitor.SERVER_HOST, "::");
+
+      // The following services will be running in the edge node host.
+      // Set the bind addresses for all of them to "${master.services.bind.address}"
+      // Which the value of `"master.services.bind.address" will be set in the AbstractProgramTwillRunnable when it
+      // starts in the remote machine
+      String masterBindAddrConf = "${" + Constants.Service.MASTER_SERVICES_BIND_ADDRESS + "}";
+
+      result.set(Constants.MessagingSystem.HTTP_SERVER_BIND_ADDRESS, masterBindAddrConf);
+      result.set(Constants.Service.MASTER_SERVICES_BIND_ADDRESS, masterBindAddrConf);
+      result.set(Constants.Dataset.Executor.ADDRESS, masterBindAddrConf);
+      result.set(Constants.Metadata.SERVICE_BIND_ADDRESS, masterBindAddrConf);
+
+      // TODO: This should probably come from the provisioner.
```

Comment:
can you open a jira for this

Reply:
Sure. Again, this is actually a path that I am not so sure how we are going to fix it, as I believe, eventually we don't want to start ZK in the cloud cluster. This is more like a temporary workaround to minimize huge architectural changes in 5.0